pkg/provider/azure_vmssflex_cache.go

```diff
@@ -123,6 +123,45 @@ func (fs *FlexScaleSet) newVmssFlexVMCache() (*azcache.TimedCache, error) {
 	return azcache.NewTimedcache(time.Duration(fs.Config.VmssFlexVMCacheTTLInSeconds)*time.Second, getter)
 }
 
+func (fs *FlexScaleSet) getNodeNameByVMName(vmName string) (string, error) {
+	fs.lockMap.LockEntry(consts.GetNodeVmssFlexIDLockKey)
+	defer fs.lockMap.UnlockEntry(consts.GetNodeVmssFlexIDLockKey)
+	cachedNodeName, isCached := fs.vmssFlexVMNameToNodeName.Load(vmName)
+	if isCached {
+		return fmt.Sprintf("%v", cachedNodeName), nil
+	}
+
+	getter := func(vmName string, crt azcache.AzureCacheReadType) (string, error) {
+		cached, err := fs.vmssFlexCache.Get(consts.VmssFlexKey, crt)
+		if err != nil {
+			return "", err
+		}
+		vmssFlexes := cached.(*sync.Map)
+
+		vmssFlexes.Range(func(key, value interface{}) bool {
+			vmssFlexID := key.(string)
+			_, err := fs.vmssFlexVMCache.Get(vmssFlexID, azcache.CacheReadTypeForceRefresh)
```

Comment:
this is also flushing cache in every `getNodeNameByVMName` call, why not add `crt azcache.AzureCacheReadType` as a parameter, and then set `azcache.CacheReadTypeDefault` as default in the caller?

Reply:
If the code goes to the line you marked, that means isCached is false in line 130, so we need to forcely refresh the cache.